build/tools/Automation/DotNet/DotNetAutomator.cs

```diff
@@ -0,0 +1,267 @@
+        private readonly IFileService fileService;
+        private readonly IYamlFileService yamlFileService;
+        private string oryxSdkStorageBaseUrl;
+        private string dotNetMinReleaseVersion;
+        private string dotNetMaxReleaseVersion;
+        private List<string> dotNetBlockedVersions = new List<string>();
+        private HashSet<string> oryxDotNetSdkVersions;
+
+        public DotNetAutomator(
+            IHttpService httpService,
+            IVersionService versionService,
+            IFileService fileService,
+            IYamlFileService yamlFileReaderService)
+        {
+            this.httpService = httpService;
+            this.versionService = versionService;
+            this.fileService = fileService;
+            this.yamlFileService = yamlFileReaderService;
+        }
+
+        public async Task RunAsync()
+        {
+            this.oryxSdkStorageBaseUrl = Environment.GetEnvironmentVariable(Constants.OryxSdkStorageBaseUrlEnvVar);
+            if (string.IsNullOrEmpty(this.oryxSdkStorageBaseUrl))
+            {
+                this.oryxSdkStorageBaseUrl = Constants.OryxSdkStorageBaseUrl;
+            }
+
+            this.oryxDotNetSdkVersions = await this.httpService.GetOryxSdkVersionsAsync(
+                Constants.OryxSdkStorageBaseUrl + DotNetConstants.DotNetSuffixUrl);
+            this.dotNetMinReleaseVersion = Environment.GetEnvironmentVariable(DotNetConstants.DotNetMinReleaseVersionEnvVar);
+            this.dotNetMaxReleaseVersion = Environment.GetEnvironmentVariable(DotNetConstants.DotNetMaxReleaseVersionEnvVar);
+            var blockedVersions = Environment.GetEnvironmentVariable(
+                DotNetConstants.DotNetBlockedVersionsEnvVar);
+            if (!string.IsNullOrEmpty(blockedVersions))
+            {
+                var versionStrings = blockedVersions.Split(',');
+                foreach (var versionString in versionStrings)
+                {
+                    this.dotNetBlockedVersions.Add(versionString.Trim());
+                }
+            }
+
+            List<DotNetVersion> newDotNetVersions = await this.GetNewDotNetVersionsAsync();
+            if (newDotNetVersions.Count > 0)
+            {
+                string constantsYamlSubPath = Path.Combine("build", Constants.ConstantsYaml);
+                List<ConstantsYamlFile> yamlConstantsObjs = await this.yamlFileService.ReadConstantsYamlFileAsync(constantsYamlSubPath);
+                this.UpdateOryxConstantsForNewVersions(newDotNetVersions, yamlConstantsObjs);
+            }
+        }
+
+        public async Task<List<DotNetVersion>> GetNewDotNetVersionsAsync()
+        {
+            List<DotNetVersion> dotNetVersions = new List<DotNetVersion>();
+
+            // Deserialize release metadata
+            var response = await this.httpService.GetDataAsync(DotNetConstants.ReleasesIndexJsonUrl);
+            var releaseNotes = response == null ? null : JsonConvert.DeserializeObject<ReleaseNotes>(response);
```

Comment:
I'm wondering if in the case where `response == null`, we should just throw an exception/exit the flow early.
Could you please clarify if there are any advantages to continue if we didn't get any reply there?

Reply:
Also, I'm assuming that this would mostly happen if the .NET team changes the path where they store the release index, right? If that's the case, how would we detect that an issue happened?

Reply:
These should never be empty. So it would make sense to throw an  exception for when they are. I've updated the code for this.